experimental/dds/tree/src/HistoryEditFactory.ts

```diff
@@ -74,9 +80,23 @@ export function revert(changes: readonly ChangeInternal[], before: RevisionView)
 				const nodesDetached = detachedNodes.get(source);
 
 				if (nodesBuilt !== undefined) {
+					if (nodesBuilt.length === 0) {
+						builtNodes.delete(source);
+						if (logger) {
+							logger.sendTelemetryEvent({ eventName: 'reverting insertion of empty traits' });
+						}
+						continue;
+					}
 					result.unshift(createInvertedInsert(change, nodesBuilt));
 					builtNodes.delete(source);
 				} else if (nodesDetached !== undefined) {
+					if (nodesDetached.length === 0) {
+						detachedNodes.delete(source);
+						if (logger) {
```

Comment:
While we don't have a lint rule for it AFAIK, our team has generally preferred to be explicit for undefined checks
`(logger !== undefined)` rather than the truthiness check `(logger)` (even though, since logger is an object, they are equivalent). 

But regardless, in this case I would recommend dropping the conditional altogether and using the optional chaining operator :)
`logger?.sendTelemetryEvent(...);`